modules/auxiliary/scanner/http/huawei_hg255s_directory_traversal.rb

```diff
@@ -0,0 +1,59 @@
+##
+# This file is part of the Metasploit Framework and may be subject to
+# redistribution and commercial restrictions. Please see the Metasploit
+# web site for more information on licensing and terms of use.
+#
+##
+
+require ‘msf/core’
+
+class Metasploit4 < Msf::Auxiliary
+   include Msf::Exploit::Remote::HttpClient
+
+   def initialize
+       super(
+           ‘Name’        => ‘Server Directory Traversal at Huawei HG255s’,
+           ‘Version’     => ‘Huawei HG255s Firmware : V100R001C163B025SP02’,
+           ‘Description’ => ‘Server Directory Traversal at Huawei HG255 by malicious GET requests’,
+           ‘Author’      => ‘Ismail Tasdelen’,
+           ‘License’     => MSF_LICENSE,
+           ‘References’     =>
+           [
+              ['CVE', '2017-17309' ],
+              ['URL', 'https://www.huawei.com/en/psirt/security-notices/huawei-sn-20170911-01-hg255s-en']
+           ]
+       )
+       register_options(
+           [
+               Opt::RPORT(80)
+           ], self.class
+       )
```

Comment:
This is redundant and can be removed. The default port for HTTP modules is TCP port `80`.

Reply:
The Huawei HG255 modem model usually uses port 80. But I saw that there are modems going through https.

Reply:
The `RPORT` option is made available automatically importing the `Msf::Exploit::Remote::HttpClient` mixin. It should not need to be explicitly defined.

If you decide to leave it in, the `self.class` can be removed, as it is no longer required.

```ruby
register_options([ Opt::RPORT(80) ])
```

Reply:
Yeah, now I get it.